### PR TITLE
Fix contacts photo integration picking wrong PHOTO entry

### PR DESCRIPTION
**Pull request by sazanof**
Hello everyone! =)

I found in 2 places that avatars are loaded incorrectly.

1) ~~In the list of recipients in the letter
**How to reproduce**: add 1 contact with the address all@example.com (in my case, this is a mailing group) and сall@example.com and set avatars for them
Write to all@example.com from your email
Open this email and see that the wrong avatar is in the recipient list. Click on "details" and see that in the list of "identical contacts" all@ and call@~~ 
Move to https://github.com/nextcloud/mail/pull/6714

2) In a message and in the list of message
**How to reproduce**: There should be 2 contacts with the same e-mail (one of them with an avatar). I have this bug when creating my adressbook + recent contacts (which appeared themselves earlier than my address book)
The problem is that if an array > 1 element can fall into the $result variable, then the first element is returned. if the photo is in the second element, the first one will be returned WITHOUT the photo.

Thank you!

It may also occur in https://github.com/nextcloud/mail/issues/4580

Signed-off-by: Mikhail Sazanov <m@sazanof.ru>

### CODE REVIEW

File: lib/Service/ContactsIntegration.php [hunk 1]
```diff
@@ -106,10 +106,12 @@ public function getMatchingRecipient(string $term): array {
 	 * @return false|null|string
 	 */
 	public function getPhoto(string $email) {
-		$result = $this->contactsManager->search($email, ['EMAIL']);
+		$result = $this->contactsManager->search($email, ['EMAIL'], ['escape_like_param' => false]);
 		if (count($result) > 0) {
```
Comment (ChristophWurst) [inline lib/Service/ContactsIntegration.php:110]: You can remove the `if` and just leave the `foreach`. This it will "skip" the loop if the array is empty.

File: lib/Service/ContactsIntegration.php [hunk 2]
```diff
@@ -106,10 +106,12 @@ public function getMatchingRecipient(string $term): array {
...
-			if (isset($result[0]['PHOTO'])) {
-				return $this->getPhotoUri($result[0]['PHOTO']);
+			foreach ($result as $contact){
+				if (isset($contact['PHOTO']) && !empty($contact['PHOTO'])) {
```
Comment (ChristophWurst) [inline lib/Service/ContactsIntegration.php:112]: Negate the condition and `continue` in the `if`, then continue with the early `return` after the `if` block.

Comment (ChristophWurst) [inline lib/Service/ContactsIntegration.php:112]: Negate the condition and `continue` in the `if`, then continue with the early `return` after the `if` block.

File: lib/Service/ContactsIntegration.php [hunk 3]
```diff
@@ -106,10 +106,12 @@ public function getMatchingRecipient(string $term): array {
 	 * @return false|null|string
 	 */
 	public function getPhoto(string $email) {
-		$result = $this->contactsManager->search($email, ['EMAIL']);
+		$result = $this->contactsManager->search($email, ['EMAIL'], ['escape_like_param' => false]);
```
Comment (ChristophWurst) [inline lib/Service/ContactsIntegration.php:109]: Could you submit this change as a separate PR? I would like to have a closer look at this. The other change can go in earlier :)

Comment (ChristophWurst) [inline lib/Service/ContactsIntegration.php:109]: Could you submit this change as a separate PR? I would like to have a closer look at this. The other change can go in earlier :)

**Review by ChristophWurst — CHANGES_REQUESTED** [review]

File: lib/Service/ContactsIntegration.php (hunk 2 above)
Comment (sazanof) [reply lib/Service/ContactsIntegration.php]: I hope I got it right

File: lib/Service/ContactsIntegration.php [hunk 4]
```diff
@@ -106,12 +106,14 @@ public function getMatchingRecipient(string $term): array {
 	 * @return false|null|string
 	 */
 	public function getPhoto(string $email) {
-		$result = $this->contactsManager->search($email, ['EMAIL']);
-		if (count($result) > 0) {
-			if (isset($result[0]['PHOTO'])) {
-				return $this->getPhotoUri($result[0]['PHOTO']);
+		$result = $this->contactsManager->search($email, ['EMAIL'], ['escape_like_param' => false]);
```
Comment (ChristophWurst) [inline lib/Service/ContactsIntegration.php:109]: shouldn't this line be part of https://github.com/nextcloud/mail/pull/6714?

Comment (sazanof) [reply lib/Service/ContactsIntegration.php:109]: I then clarified in the chat about the separation of pr (`getPhoto` In one PR, `doSearch` in another PR).

 Maybe we didn't understand each other? You probably wanted to "have a closer look at" `['escape_like_param' => false]` parameter ?

Comment (ChristophWurst) [reply lib/Service/ContactsIntegration.php:109]: > Maybe we didn't understand each other? You probably wanted to "have a closer look at" `['escape_like_param' => false]` parameter ?

Exactly. I want to be sure the turned off escaping doesn't have unexpected side effects

**Review by ChristophWurst — APPROVED** [review]

File: lib/Service/ContactsIntegration.php [hunk 5]
```diff
@@ -107,11 +107,13 @@ public function getMatchingRecipient(string $term): array {
 	 */
 	public function getPhoto(string $email) {
 		$result = $this->contactsManager->search($email, ['EMAIL']);
-		if (count($result) > 0) {
-			if (isset($result[0]['PHOTO'])) {
-				return $this->getPhotoUri($result[0]['PHOTO']);
+		foreach ($result as $contact){
```
Comment (ChristophWurst) [inline lib/Service/ContactsIntegration.php:110]:
```suggestion
		foreach ($result as $contact) {
```

Comment (sazanof) [reply lib/Service/ContactsIntegration.php]: Fixed. Space added.

**Review by ChristophWurst — APPROVED** [review]
